src/store/actions.js

```diff
@@ -149,10 +146,15 @@ export default {
     resetStore(watchAssetManager.store, unapprovedAssetMsgsHandler)
     assetDetectionController.stopAssetDetection()
     // torus.updateStaticData({ isUnlocked: false })
-    if (isMain && selectedAddress) {
-      router.push({ path: '/logout' }).catch(() => {})
+    if (selectedAddress) {
+      const openLoginHandler = OpenLoginHandler.getInstance()
+      if (isMain) {
+        router.push({ path: '/logout' }).catch(() => {})
+      } else {
+        // optimistically removing sessionId.
+        openLoginHandler.openLoginInstance.state.store.set('sessionId', null)
+      }
       try {
-        const openLoginHandler = OpenLoginHandler.getInstance()
         await openLoginHandler.invalidateSession()
```

Comment:
invalidate session removes sessionID from store already

Reply:
if sessionId is invalid, then  invalidateSession won't remove sessionId from store coz it will throw.

Reply:
added the logic in invalidateSession itself